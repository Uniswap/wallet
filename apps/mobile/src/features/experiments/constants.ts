/**
 * Feature flag names
 * These should match the Gate Key on Statsig
 */
export enum FEATURE_FLAGS {
  FiatOnRamp = 'fiat-onramp',
  SwapPermit2 = 'swap-permit-2',
  WalletConnectV2 = 'wc-v2',
}

/**
 * Dynamic Configs
 * These should match the dynamic config's `Config Name` on Statsig
 * https://console.statsig.com/5M2TFMQiHkbY9RML95FAEa/dynamic_configs
 */
export enum DYNAMIC_CONFIGS {
  ForceUpgrade = 'force_upgrade',
}
