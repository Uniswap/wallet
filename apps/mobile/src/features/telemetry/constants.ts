import { RootParamList } from 'src/app/navigation/types'
import { AppScreen, Screens } from 'src/screens/Screens'

export function getAuthMethod(
  isSettingEnabled: boolean,
  isTouchIdSupported: boolean,
  isFaceIdSupported: boolean
): AuthMethod {
  if (!isSettingEnabled) return AuthMethod.None

  // both cannot be true since no iOS device supports both
  if (isFaceIdSupported) return AuthMethod.FaceId
  if (isTouchIdSupported) return AuthMethod.TouchId

  return AuthMethod.None
}

export function getEventParams(
  screen: AppScreen,
  params: RootParamList[AppScreen]
): Record<string, unknown> | undefined {
  switch (screen) {
    case Screens.SettingsWallet:
      return {
        address: (params as RootParamList[Screens.SettingsWallet]).address,
      }
    case Screens.SettingsWalletEdit:
      return {
        address: (params as RootParamList[Screens.SettingsWalletEdit]).address,
      }
    default:
      return undefined
  }
}

/**
 * Event names that occur in this specific application
 */
export enum MobileEventName {
  BalancesReport = 'Balances Report',
  DeepLinkOpened = 'Deep Link Opened',
  ExploreFilterSelected = 'Explore Filter Selected',
  ExploreSearchResultClicked = 'Explore Search Result Clicked',
  ExploreSearchCancel = 'Explore Search Cancel',
  ExploreTokenItemSelected = 'Explore Token Item Selected',
  FavoriteItem = 'Favorite Item',
  FiatOnRampBannerPressed = 'Fiat OnRamp Banner Pressed',
  FiatOnRampQuickActionButtonPressed = 'Fiat OnRamp QuickAction Button Pressed',
  FiatOnRampAmountEntered = 'Fiat OnRamp Amount Entered',
  FiatOnRampWidgetOpened = 'Fiat OnRamp Widget Opened',
  NetworkFilterSelected = 'Network Filter Selected',
  OnboardingCompleted = 'Onboarding Completed',
  PerformanceReport = 'Performance Report',
  PerformanceGraphql = 'Performance GraphQL',
  PortfolioBalanceFreshnessLag = 'Portfolio Balance Freshness Lag',
  SwapSubmitted = 'Swap Submitted to Provider',
  TokenDetailsOtherChainButtonPressed = 'Token Details Other Chain Button Pressed',
  TokenSelected = 'Token Selected',
  WalletAdded = 'Wallet Added',
  WalletConnectSheetCompleted = 'Wallet Connect Sheet Completed',
  // alphabetize additional values.
}

/**
 * Possible names for the section property in TraceContext
 */
export const enum SectionName {
  CurrencyInputPanel = 'currency-input-panel',
  CurrencyOutputPanel = 'currency-output-panel',
  ExploreFavoriteTokensSection = 'explore-favorite-tokens-section',
  ExploreSearch = 'explore-search',
  ExploreTopTokensSection = 'explore-top-tokens-section',
  HomeActivityTab = 'home-activity-tab',
  HomeNFTsTab = 'home-nfts-tab',
  HomeTokensTab = 'home-tokens-tab',
  ImportAccountForm = 'import-account-form',
  ProfileActivityTab = 'profile-activity-tab',
  ProfileNftsTab = 'profile-nfts-tab',
  ProfileTokensTab = 'profile-tokens-tab',
  Sidebar = 'sidebar',
  SwapForm = 'swap-form',
  SwapPending = 'swap-pending',
  SwapReview = 'swap-review',
  TokenSelector = 'token-selector',
  TransferForm = 'transfer-form',
  TransferPending = 'transfer-pending',
  TransferReview = 'transfer-review',
  // alphabetize additional values.
}

/**
 * Possible names for the modal property in TraceContext
 */
export const enum ModalName {
  AccountEdit = 'account-edit-modal',
  AccountSwitcher = 'account-switcher-modal',
  AddWallet = 'add-wallet-modal',
  BlockedAddress = 'blocked-address',
  Experiments = 'experiments',
  Explore = 'explore-modal',
  FaceIDWarning = 'face-id-warning',
  FiatOnRamp = 'fiat-on-ramp',
  ForceUpgradeModal = 'force-upgrade-modal',
  ICloudBackupInfo = 'icloud-backup-info-modal',
  NetworkSelector = 'network-selector-modal',
  NftCollection = 'nft-collection',
  RecoveryWarning = 'recovery-warning',
  RemoveWallet = 'remove-wallet-modal',
  RestoreWallet = 'restore-wallet-modal',
  RemoveSeedPhraseWarningModal = 'remove-seed-phrase-warning-modal',
  ReimportUninstall = 'reimport-uninstall-modal',
  ScreenshotWarning = 'screenshot-warning',
  Send = 'send-modal',
  SeedPhraseWarningModal = 'seed-phrase-warning-modal',
  SendWarning = 'send-warning-modal',
  SlippageInfo = 'slippage-info-modal',
  Swap = 'swap-modal',
  SwapSettings = 'swap-settings-modal',
  SwapWarning = 'swap-warning-modal',
  GasEstimateWarning = 'gas-estimate-warning-modal',
  TokenSelector = 'token-selector',
  TokenWarningModal = 'token-warning-modal',
  TooltipContent = 'tooltip-content',
  TransactionActions = 'transaction-actions',
  ViewSeedPhraseWarning = 'view-seed-phrase-warning',
  WalletConnectScan = 'wallet-connect-scan-modal',
  WCDappConnectedNetworks = 'wc-dapp-connected-networks-modal',
  WCPendingConnection = 'wc-pending-connection-modal',
  WCSignRequest = 'wc-sign-request-modal',
  WCSwitchChainRequest = 'wc-switch-chain-request-modal',
  WCViewOnlyWarning = 'wc-view-only-warning-modal',
  // alphabetize additional values.
}

/**
 * Views not within the navigation stack that we still want to
 * log Pageview events for. (Usually presented as nested views within another screen)
 */
export const enum ManualPageViewScreen {
  WriteDownRecoveryPhrase = 'WriteDownRecoveryPhrase',
  ConfirmRecoveryPhrase = 'ConfirmRecoveryPhrase',
}

/**
 * Possible names for the element property in TraceContext
 */
export const enum ElementName {
  AcceptNewRate = 'accept-new-rate',
  AccountCard = 'account-card',
  AddManualBackup = 'add-manual-backup',
  AddViewOnlyWallet = 'add-view-only-wallet',
  AddiCloudBackup = 'add-icloud-backup',
  AddBackupNone = 'add-backup-none',
  Back = 'back',
  Buy = 'buy',
  Cancel = 'cancel',
  Confirm = 'confirm',
  Continue = 'continue',
  Copy = 'copy',
  CreateAccount = 'create-account',
  Disconnect = 'disconnect',
  Edit = 'edit',
  EmptyStateBuy = 'empty-state-buy',
  EmptyStateGetStarted = 'empty-state-get-started',
  EmptyStateImport = 'empty-state-get-import',
  EmptyStateReceive = 'empty-state-receive',
  Enable = 'enable',
  EtherscanView = 'etherscan-view',
  Favorite = 'favorite',
  FiatOnRampTokenSelector = 'fiat-on-ramp-token-selector',
  FiatOnRampWidgetButton = 'fiat-on-ramp-widget-button',
  GetHelp = 'get-help',
  GetStarted = 'get-started',
  ImportAccount = 'import',
  Manage = 'manage',
  ManageConnections = 'manage-connections',
  MoonpayExplorerView = 'moonpay-explorer-view',
  NetworkButton = 'network-button',
  Next = 'next',
  Notifications = 'notifications',
  OK = 'ok',
  OnboardingCreateWallet = 'onboarding-create-wallet',
  OnboardingImportBackup = 'onboarding-import-backup',
  OnboardingImportSeedPhrase = 'onboarding-import-seed-phrase',
  OnboardingImportWallet = 'onboarding-import-wallet',
  OnboardingImportWatchedAccount = 'onboarding-import-watched-account',
  OpenSettingsButton = 'open-settings-button',
  QRCodeModalToggle = 'qr-code-modal-toggle',
  Receive = 'receive',
  RecoveryHelpButton = 'recovery-help-button',
  Remove = 'remove',
  RestoreFromICloud = 'restore-from-icloud',
  RestoreWallet = 'restore-wallet',
  ReviewSwap = 'review-swap',
  ReviewTransfer = 'review-transfer',
  SearchEtherscanItem = 'search-etherscan-item',
  SearchNFTCollectionItem = 'search-nft-collection-item',
  SearchTokenItem = 'search-token-item',
  SearchWalletItem = 'search-wallet-item',
  SelectColor = 'select-color',
  SelectRecipient = 'select-recipient',
  Send = 'send',
  SetMax = 'set-max',
  Settings = 'settings',
  Skip = 'skip',
  Submit = 'submit',
  Swap = 'swap',
  SwapSettings = 'swap-settings',
  SwitchCurrenciesButton = 'switch-currencies-button',
  TimeFrame1H = 'time-frame-1H',
  TimeFrame1D = 'time-frame-1D',
  TimeFrame1W = 'time-frame-1W',
  TimeFrame1M = 'time-frame-1M',
  TimeFrame1Y = 'time-frame-1Y',
  TokenAddress = 'token-address',
  TokenInputSelector = 'token-input-selector',
  TokenLinkEtherscan = 'token-link-etherscan',
  TokenLinkTwitter = 'token-link-twitter',
  TokenLinkWebsite = 'token-link-website',
  TokenOutputSelector = 'token-output-selector',
  TokenSelectorToggle = 'token-selector-toggle',
  TokenWarningAccept = 'token-warning-accept',
  Unwrap = 'unwrap',
  WCDappSwitchAccount = 'wc-dapp-switch-account',
  WCDappSwitchNetwork = 'wc-dapp-switch-network',
  WCOpenDapp = 'wc-open-dapp',
  WalletCard = 'wallet-card',
  WalletConnectScan = 'wallet-connect-scan',
  WalletQRCode = 'wallet-qr-code',
  WalletSettings = 'WalletSettings',
  Wrap = 'wrap',
  // alphabetize additional values.
}

/**
 * User properties tied to user rather than events
 */
export enum UserPropertyName {
  ActiveWalletAddress = 'active_wallet_address',
  ActiveWalletType = 'active_wallet_type',
  AppOpenAuthMethod = 'app_open_auth_method',
  AppVersion = 'app_version',
  DarkMode = 'is_dark_mode',
  IsCloudBackedUp = 'is_cloud_backed_up',
  IsHideSmallBalancesEnabled = 'is_hide_small_balances_enabled',
  IsHideSpamTokensEnabled = 'is_hide_spam_tokens_enabled',
  IsPushEnabled = 'is_push_enabled',
  TransactionAuthMethod = 'transaction_auth_method',
  WalletSignerAccounts = `wallet_signer_accounts`,
  WalletSignerCount = 'wallet_signer_count',
  WalletViewOnlyCount = 'wallet_view_only_count',
  // alphabetize additional values.
}

export enum AuthMethod {
  FaceId = 'FaceId',
  None = 'None',
  TouchId = 'TouchId',
  // alphabetize additional values.
}
