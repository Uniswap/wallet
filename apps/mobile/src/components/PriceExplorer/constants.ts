import { HistoryDuration } from 'src/data/__generated__/types-and-hooks'
import { ElementName } from 'src/features/telemetry/constants'
import { dimensions } from 'src/styles/sizing'
import { theme as FixedTheme } from 'src/styles/theme'

// sets the height of the chart short enough on small devices that the
// "Your balance" section will always show above the fold
// we can't use useResponsiveProps for this because CHART_HEIGHT gets
// used in non-component code related to chart functionality
export const CHART_HEIGHT = dimensions.fullHeight < FixedTheme.breakpoints.sm.height ? 180 : 264
export const CHART_WIDTH = dimensions.fullWidth

export const NUM_GRAPHS = 5

export const BUTTON_PADDING = 20
export const BUTTON_WIDTH = CHART_WIDTH / NUM_GRAPHS
export const LABEL_WIDTH = BUTTON_WIDTH - BUTTON_PADDING * 2

export const CURSOR_INNER_SIZE = 12
export const CURSOR_SIZE = CURSOR_INNER_SIZE + 6
export const LINE_WIDTH = 1

export const TIME_RANGES = [
  [HistoryDuration.Hour, '1H', ElementName.TimeFrame1H],
  [HistoryDuration.Day, '1D', ElementName.TimeFrame1D],
  [HistoryDuration.Week, '1W', ElementName.TimeFrame1W],
  [HistoryDuration.Month, '1M', ElementName.TimeFrame1M],
  [HistoryDuration.Year, '1Y', ElementName.TimeFrame1Y],
] as const
