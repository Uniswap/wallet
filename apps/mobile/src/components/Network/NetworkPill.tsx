import React, { ComponentProps } from 'react'
import { Pill } from 'src/components/text/Pill'
import { useNetworkColors } from 'src/utils/colors'
import { iconSizes } from 'ui/src/theme/iconSizes'
import { NetworkLogo } from 'wallet/src/components/CurrencyLogo/NetworkLogo'
import { ChainId, CHAIN_INFO } from 'wallet/src/constants/chains'

export type NetworkPillProps = {
  chainId: ChainId
  showBackgroundColor?: boolean
  showBorder?: boolean
  showIcon?: boolean
} & ComponentProps<typeof Pill>

export function NetworkPill({
  chainId,
  showBackgroundColor = true,
  showBorder,
  showIcon = false,
  ...rest
}: NetworkPillProps): JSX.Element {
  const info = CHAIN_INFO[chainId]
  const colors = useNetworkColors(chainId)

  return (
    <Pill
      customBackgroundColor={showBackgroundColor ? colors?.background : undefined}
      customBorderColor={showBorder ? colors.foreground : 'transparent'}
      foregroundColor={colors.foreground}
      icon={showIcon ? <NetworkLogo chainId={chainId} size={iconSizes.icon16} /> : null}
      label={info.label}
      {...rest}
    />
  )
}

export function InlineNetworkPill(props: NetworkPillProps): JSX.Element {
  return (
    <NetworkPill
      borderRadius="rounded12"
      px="spacing8"
      py="spacing2"
      textVariant="buttonLabelMicro"
      {...props}
    />
  )
}
