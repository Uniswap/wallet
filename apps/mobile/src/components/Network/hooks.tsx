import { default as React, useMemo } from 'react'
import { useAppTheme } from 'src/app/hooks'
import Check from 'src/assets/icons/check.svg'
import { NetworkLogo } from 'src/components/CurrencyLogo/NetworkLogo'
import { Flex } from 'src/components/layout'
import { Box } from 'src/components/layout/Box'
import { Separator } from 'src/components/layout/Separator'
import { Text } from 'src/components/Text'
import { ChainId, CHAIN_INFO } from 'src/constants/chains'
import { useActiveChainIds } from 'src/features/chains/utils'
import { ElementName } from 'src/features/telemetry/constants'
import { iconSizes } from 'src/styles/sizing'

export function useNetworkOptions(
  selectedChain: ChainId | null,
  onPress: (chainId: ChainId | null) => void
): { key: string; onPress: () => void; render: () => JSX.Element }[] {
  const theme = useAppTheme()
  const activeChains = useActiveChainIds()

  return useMemo(
    () =>
      activeChains.map((chainId) => {
        const info = CHAIN_INFO[chainId]
        return {
          key: `${ElementName.NetworkButton}-${chainId}`,
          onPress: () => onPress(chainId),
          render: () => (
            <>
              <Separator />
              <Flex
                row
                alignItems="center"
                justifyContent="space-between"
                px="spacing24"
                py="spacing16">
                <NetworkLogo chainId={chainId} size={iconSizes.icon24} />
                <Text color="textPrimary" variant="bodyLarge">
                  {info.label}
                </Text>
                <Box height={iconSizes.icon20} width={iconSizes.icon20}>
                  {selectedChain === chainId && (
                    <Check
                      color={theme.colors.accentActive}
                      height={iconSizes.icon20}
                      width={iconSizes.icon20}
                    />
                  )}
                </Box>
              </Flex>
            </>
          ),
        }
      }),
    [activeChains, onPress, selectedChain, theme.colors.accentActive]
  )
}
