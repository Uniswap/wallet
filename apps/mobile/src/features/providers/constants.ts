import { ChainId, ChainIdTo } from 'src/constants/chains'

export interface FlashbotsInfo {
  rpcUrl: string
  txApi: string
}

// enable Goerli once Flashbot's Transaction status API is available there
export const FLASHBOTS_URLS: ChainIdTo<FlashbotsInfo> = {
  [ChainId.Mainnet]: {
    rpcUrl: 'https://rpc.flashbots.net',
    txApi: 'https://protect.flashbots.net/tx',
  },
}
