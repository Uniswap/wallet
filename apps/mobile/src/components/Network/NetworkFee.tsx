import React from 'react'
import { useTranslation } from 'react-i18next'
import { useAppTheme } from 'src/app/hooks'
import InfoCircleSVG from 'src/assets/icons/info-circle.svg'
import { TouchableArea } from 'src/components/buttons/TouchableArea'
import { Box, Flex } from 'src/components/layout'
import { SpinningLoader } from 'src/components/loading/SpinningLoader'
import { InlineNetworkPill } from 'src/components/Network/NetworkPill'
import { Text } from 'src/components/Text'
import { ChainId } from 'src/constants/chains'
import { useUSDValue } from 'src/features/gas/hooks'
import { formatUSDPrice, NumberType } from 'src/utils/format'

export function NetworkFee({
  chainId,
  gasFee,
  gasFallbackUsed,
  onShowGasWarning,
}: {
  chainId: ChainId
  gasFee?: string
  gasFallbackUsed?: boolean
  onShowGasWarning?: () => void
}): JSX.Element {
  const { t } = useTranslation()
  const theme = useAppTheme()
  const gasFeeUSD = useUSDValue(chainId, gasFee)

  const feeSectionContent = (
    <>
      <Text
        color={gasFallbackUsed && gasFeeUSD ? 'accentWarning' : 'textPrimary'}
        variant="subheadSmall">
        {formatUSDPrice(gasFeeUSD, NumberType.FiatGasPrice)}
      </Text>
      {gasFallbackUsed && gasFeeUSD && (
        <Box ml="spacing4">
          <InfoCircleSVG
            color={theme.colors.accentWarning}
            height={theme.iconSizes.icon20}
            width={theme.iconSizes.icon20}
          />
        </Box>
      )}
    </>
  )

  return (
    <Flex row alignItems="center" justifyContent="space-between">
      <Text variant="subheadSmall">{t('Network fee')}</Text>
      <Flex row alignItems="center" gap="spacing8">
        <InlineNetworkPill chainId={chainId} />
        <Text variant="subheadSmall">â€¢</Text>
        {!gasFeeUSD ? (
          <SpinningLoader size={theme.iconSizes.icon20} />
        ) : gasFallbackUsed && onShowGasWarning ? (
          <TouchableArea
            alignItems="center"
            flexDirection="row"
            justifyContent="space-between"
            onPress={onShowGasWarning}>
            {feeSectionContent}
          </TouchableArea>
        ) : (
          <Flex row alignItems="center" justifyContent="space-between">
            {feeSectionContent}
          </Flex>
        )}
      </Flex>
    </Flex>
  )
}
