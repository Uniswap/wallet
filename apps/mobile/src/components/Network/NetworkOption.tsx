import { default as React } from 'react'
import { useTranslation } from 'react-i18next'
import { useAppTheme } from 'src/app/hooks'
import { Box, Flex } from 'src/components/layout'
import { Separator } from 'src/components/layout/Separator'
import { Text } from 'src/components/Text'
import Check from 'ui/src/assets/icons/check.svg'
import { iconSizes } from 'ui/src/theme/iconSizes'
import { NetworkLogo } from 'wallet/src/components/CurrencyLogo/NetworkLogo'
import { ChainId, CHAIN_INFO } from 'wallet/src/constants/chains'

const NETWORK_OPTION_ICON_SIZE = iconSizes.icon24

export function NetworkOption({
  chainId,
  currentlySelected,
}: {
  chainId: ChainId | null
  currentlySelected?: boolean
}): JSX.Element {
  const { t } = useTranslation()
  const theme = useAppTheme()
  const info = chainId && CHAIN_INFO[chainId]
  return (
    <>
      <Separator />
      <Flex row alignItems="center" justifyContent="space-between" px="spacing24" py="spacing16">
        {(chainId && <NetworkLogo chainId={chainId} size={NETWORK_OPTION_ICON_SIZE} />) || (
          <Box width={NETWORK_OPTION_ICON_SIZE} />
        )}
        <Text color="textPrimary" variant="bodyLarge">
          {info?.label ?? t('All networks')}
        </Text>
        <Flex centered height={NETWORK_OPTION_ICON_SIZE} width={NETWORK_OPTION_ICON_SIZE}>
          {currentlySelected && (
            <Check
              color={theme.colors.textPrimary}
              height={iconSizes.icon20}
              width={iconSizes.icon20}
            />
          )}
        </Flex>
      </Flex>
    </>
  )
}
