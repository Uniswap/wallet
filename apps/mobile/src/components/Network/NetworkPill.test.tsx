import React from 'react'
import { InlineNetworkPill, NetworkPill } from 'src/components/Network/NetworkPill'
import { ChainId } from 'src/constants/chains'
import { render } from 'src/test/test-utils'

describe(NetworkPill, () => {
  it('renders a NetworkPill without image', () => {
    const tree = render(<NetworkPill chainId={ChainId.Goerli} />)
    expect(tree).toMatchSnapshot()
  })

  it('renders a NetworkPill with border', () => {
    const tree = render(<NetworkPill chainId={ChainId.Goerli} showBorder={true} />)
    expect(tree).toMatchSnapshot()
  })

  it('renders an InlineNetworkPill', () => {
    const tree = render(<InlineNetworkPill chainId={ChainId.Goerli} />)
    expect(tree).toMatchSnapshot()
  })
})
