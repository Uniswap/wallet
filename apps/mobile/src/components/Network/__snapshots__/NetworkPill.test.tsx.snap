// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetworkPill renders a NetworkPill with border 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderColor": "transparent",
        "borderRadius": 999999,
        "borderWidth": 1,
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingHorizontal": 4,
        "paddingVertical": 8,
      },
      Object {
        "backgroundColor": "#2098531a",
        "borderColor": "#209853",
      },
    ]
  }
>
  <Text
    allowFontScaling={true}
    maxFontSizeMultiplier={1.4}
    style={
      Array [
        Object {
          "color": "#0E1524",
          "display": "flex",
          "fontFamily": "Inter-Regular",
          "fontSize": 15,
          "lineHeight": 20,
        },
        Object {
          "color": "#209853",
        },
      ]
    }
  >
    Görli
  </Text>
</View>
`;

exports[`NetworkPill renders a NetworkPill without image 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderColor": "transparent",
        "borderRadius": 999999,
        "borderWidth": 1,
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingHorizontal": 4,
        "paddingVertical": 8,
      },
      Object {
        "backgroundColor": "#2098531a",
        "borderColor": "transparent",
      },
    ]
  }
>
  <Text
    allowFontScaling={true}
    maxFontSizeMultiplier={1.4}
    style={
      Array [
        Object {
          "color": "#0E1524",
          "display": "flex",
          "fontFamily": "Inter-Regular",
          "fontSize": 15,
          "lineHeight": 20,
        },
        Object {
          "color": "#209853",
        },
      ]
    }
  >
    Görli
  </Text>
</View>
`;

exports[`NetworkPill renders an InlineNetworkPill 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderColor": "transparent",
        "borderRadius": 12,
        "borderWidth": 1,
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingHorizontal": 8,
        "paddingVertical": 2,
      },
      Object {
        "backgroundColor": "#2098531a",
        "borderColor": "transparent",
      },
    ]
  }
>
  <Text
    allowFontScaling={true}
    maxFontSizeMultiplier={1.2}
    style={
      Array [
        Object {
          "color": "#0E1524",
          "display": "flex",
          "fontFamily": "Inter-SemiBold",
          "fontSize": 12,
          "lineHeight": 17,
        },
        Object {
          "color": "#209853",
        },
      ]
    }
  >
    Görli
  </Text>
</View>
`;
